allennlp/tests/data/dataset_readers/semantic_parsing/template_text2sql_test.py

```diff
@@ -0,0 +1,63 @@
+# pylint: disable=no-self-use,invalid-name,line-too-long
+
+from allennlp.data.dataset_readers import TemplateText2SqlDatasetReader
+from allennlp.common.util import ensure_list
+from allennlp.common.testing import AllenNlpTestCase
+
+
+class TestTemplateText2SqlDatasetReader(AllenNlpTestCase):
+    def test_reader(self):
+        reader = TemplateText2SqlDatasetReader()
+
+        instances = reader.read(str(AllenNlpTestCase.FIXTURES_ROOT / "data" / "text2sql" / "*"))
+        instances = ensure_list(instances)
+
+        fields = instances[0].fields
+        tokens = [t.text for t in fields["tokens"].tokens]
+        tags = fields["slot_tags"].labels
+
+        assert tokens == ['how', 'many', 'buttercup', 'kitchen', 'are', 'there', 'in', 'san', 'francisco', '?']
+        assert tags == ['O', 'O', 'name0', 'name0', 'O', 'O', 'O', 'city_name0', 'city_name0', 'O']
+        assert fields["template"].label == "SELECT COUNT ( * ) FROM LOCATION AS LOCATIONalias0 , RESTAURANT " \
```

Comment:
So you predict what the template is and also run some kind of CRF tagger to fill in the variables in the template?  Do you constrain the tagger to only use the variables in the template?

Reply:
Right, exactly - no, in the text2sql paper constraints are not considered. That would be a good and easy extension.